public/app/core/reducers/navBarTree.ts

```diff
@@ -51,6 +51,29 @@ const navTreeSlice = createSlice({
         }
       }
     },
+    setBookmark: (state, action: PayloadAction<{ item: NavModelItem; isSaved: boolean }>) => {
+      const bookmarks = state.find((navItem) => navItem.id === 'bookmarks');
+      const { item, isSaved } = action.payload;
+      if (bookmarks && config.featureToggles.pinNavItems) {
```

Comment:
Should we separate the toggle condition, so it's easier to remove in the future? e.g.,
```tsx
if (!config.featureToggles.pinNavItems) {
     return;
}
```
at the top of the function?